extras/rya.geoindexing/src/main/java/org/apache/rya/indexing/geotemporal/GeoTemporalIndexSetProvider.java

```diff
@@ -0,0 +1,251 @@
+        } else {
+            return null;
+        }
+    }
+
+    private FUNCTION_TYPE ensureSameType(final Collection<IndexingExpr> filters) {
+        FUNCTION_TYPE type = null;
+        for(final IndexingExpr filter : filters) {
+            if(type == null) {
+                type = IndexingFunctionRegistry.getFunctionType(filter.getFunction());
+            } else {
+                if(IndexingFunctionRegistry.getFunctionType(filter.getFunction()) != type) {
+                    return null;
+                }
+            }
+        }
+        return type;
+    }
+
+    private void buildMaps(final QuerySegment<EventQueryNode> node) {
+        final List<QueryModelNode> unused = new ArrayList<>();
+        for (final QueryModelNode pattern : node.getOrderedNodes()) {
+            if(pattern instanceof FunctionCall) {
+                discoverFilter((FunctionCall) pattern, unused);
+            }
+            if(pattern instanceof StatementPattern) {
+                discoverPatterns((StatementPattern) pattern, unused);
+            }
+        }
+    }
+
+    private void discoverFilter(final FunctionCall filter, final List<QueryModelNode> unmatched) {
+        try {
+            filter.visit(new FilterVisitor());
+        } catch (final Exception e) {
+            e.printStackTrace();
+        }
+    }
+
+    private void discoverPatterns(final StatementPattern pattern, final List<QueryModelNode> unmatched) {
+        final Var subj = pattern.getSubjectVar();
+        final Var objVar = pattern.getObjectVar();
+
+        patternMap.put(subj, pattern);
+        objectPatterns.put(objVar, pattern);
+        //check for existing filters.
+        if(unmatchedFilters.containsKey(objVar)) {
+            final Collection<FunctionCall> calls = unmatchedFilters.removeAll(objVar);
+            for(final FunctionCall call : calls) {
+                addFilter(call);
+                matchedFilters.put(objVar, call);
+            }
+        }
+    }
+
+    @Override
+    public Iterator<List<EventQueryNode>> getExternalSetCombos(final QuerySegment<EventQueryNode> segment) {
+        final List<List<EventQueryNode>> comboList = new ArrayList<>();
+        comboList.add(getExternalSets(segment));
+        return comboList.iterator();
+    }
+
+    private void addFilter(final FunctionCall call) {
+        filterURI = new URIImpl(call.getURI());
+        final Var objVar = IndexingFunctionRegistry.getResultVarFromFunctionCall(filterURI, call.getArgs());
+        filterMap.put(objVar, new IndexingExpr(filterURI, objectPatterns.get(objVar), extractArguments(objVar.getName(), call)));
+    }
+
+    private Value[] extractArguments(final String matchName, final FunctionCall call) {
+        final Value args[] = new Value[call.getArgs().size() - 1];
+        int argI = 0;
+        for (int i = 0; i != call.getArgs().size(); ++i) {
```

Comment:
this was extracted from the geoindexer before the projects were merged.  I'll extract where this is called from in geo and have both use this.

I didn't write this